### ISSUE TITLE
Error in rep.int(NA_character_, max(ends - 1)) : invalid 'times' value

### ISSUE THREAD

**Issue by IndrajeetPatil** [jimhester/lintr#355]
`lintr::lint_package()` is not working for me.

Here is the call stack-

```r
> lintr::lint_package()
..................Error in rep.int(NA_character_, max(ends - 1)) : invalid 'times' value
In addition: Warning message:
In max(ends - 1) : no non-missing arguments to max; returning -Inf

Enter a frame number, or 0 to exit   

 1: lintr::lint_package()
 2: flatten_lints(lapply(files, function(file) {
    if (interactive()) {
        message(".", appendLF = FALSE)
    }
    lin
 3: structure(flatten_list(x, class = "lint"), class = "lints")
 4: flatten_list(x, class = "lint")
 5: assign_item(x)
 6: lapply(files, function(file) {
    if (interactive()) {
        message(".", appendLF = FALSE)
    }
    lint(file, ..., p
 7: FUN(X[[i]], ...)
 8: lint(file, ..., parse_settings = FALSE)
 9: get_source_expressions(filename)
10: extract_r_source(source_file$filename, source_file$lines)

Selection: 
```

My config file for `lintr`:
```
linters: with_defaults(object_name_linter=NULL,line_length_linter(100),trailing_whitespace_linter=NULL)
```

Session information:
``` r
options(width = 200)
devtools::session_info(include_base = TRUE)
#> - Session info ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
#>  setting  value                                             
#>  version  R Under development (unstable) (2018-10-20 r75474)
#>  os       Windows 10 x64                                    
#>  system   x86_64, mingw32                                   
#>  ui       RTerm                                             
#>  language (EN)                                              
#>  collate  English_United States.1252                        
#>  ctype    English_United States.1252                        
#>  tz       America/New_York                                  
#>  date     2018-11-12                                        
#> 
#> - Packages -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
#>  ! package     * version    date       lib source                            
#>    assertthat    0.2.0      2017-04-11 [1] CRAN (R 3.5.1)                    
#>    backports     1.1.2      2017-12-13 [1] CRAN (R 3.5.0)                    
#>    base        * 3.6.0      2018-10-21 [?] local                             
#>    base64enc     0.1-3      2015-07-28 [1] CRAN (R 3.5.0)                    
#>    callr         3.0.0.9001 2018-10-30 [1] Github (r-lib/callr@942ba24)      
#>    cli           1.0.1.9000 2018-10-30 [1] Github (r-lib/cli@56538e3)        
#>  P compiler      3.6.0      2018-10-21 [2] local                             
#>    crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.1)                    
#>  P datasets    * 3.6.0      2018-10-21 [2] local                             
#>    debugme       1.1.0      2017-10-22 [1] CRAN (R 3.5.1)                    
#>    desc          1.2.0      2018-10-30 [1] Github (r-lib/desc@7c12d36)       
#>    devtools      2.0.1      2018-10-26 [1] CRAN (R 3.5.1)                    
#>    digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.1)                    
#>    evaluate      0.12       2018-10-09 [1] CRAN (R 3.5.1)                    
#>    fs            1.2.6      2018-08-23 [1] CRAN (R 3.5.1)                    
#>    glue          1.3.0      2018-07-17 [1] CRAN (R 3.5.1)                    
#>  P graphics    * 3.6.0      2018-10-21 [2] local                             
#>  P grDevices   * 3.6.0      2018-10-21 [2] local                             
#>    htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.1)                    
#>    knitr         1.20       2018-02-20 [1] CRAN (R 3.5.1)                    
#>    magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.1)                    
#>    memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.1)                    
#>  P methods     * 3.6.0      2018-10-21 [2] local                             
#>    pkgbuild      1.0.2      2018-10-16 [1] CRAN (R 3.5.1)                    
#>    pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.6.0)                    
#>    prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.1)                    
#>    processx      3.2.0      2018-08-16 [1] CRAN (R 3.5.1)                    
#>    ps            1.2.1      2018-11-06 [1] CRAN (R 3.6.0)                    
#>    R6            2.3.0      2018-10-04 [1] CRAN (R 3.5.1)                    
#>    Rcpp          1.0.0      2018-11-07 [1] CRAN (R 3.6.0)                    
#>    remotes       2.0.2      2018-10-30 [1] CRAN (R 3.6.0)                    
#>    rlang         0.3.0.1    2018-10-25 [1] CRAN (R 3.5.1)                    
#>    rmarkdown     1.10.15    2018-11-08 [1] Github (rstudio/rmarkdown@9875136)
#>    rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.1)                    
#>    sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.6.0)                    
#>  P stats       * 3.6.0      2018-10-21 [2] local                             
#>    stringi       1.2.4      2018-07-20 [1] CRAN (R 3.6.0)                    
#>    stringr       1.3.1      2018-05-10 [1] CRAN (R 3.5.1)                    
#>    testthat      2.0.1      2018-10-13 [1] CRAN (R 3.5.1)                    
#>  P tools         3.6.0      2018-10-21 [2] local                             
#>    usethis       1.4.0.9000 2018-10-28 [1] Github (r-lib/usethis@44dd72c)    
#>  P utils       * 3.6.0      2018-10-21 [2] local                             
#>    withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.1)                    
#>    yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.1)                    
#> 
#> [1] C:/Users/inp099/Documents/R/win-library/3.6
#> [2] C:/Program Files/R/R-devel/library
#> 
#>  P -- Loaded and on-disk path mismatch.
```

<sup>Created on 2018-11-12 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by russHyde** [comment]
Could you confirm the lintr version you were using at the time, it is absent from the `sessionInfo`. Also could you confirm whether you have any .Rmd, .Rnw etc files in your package.

**Comment by IndrajeetPatil** [comment]
Here are the needed details:
```r
# package version
packageVersion("lintr")
[1] ‘1.0.3’

# rmarkdown files
list.files(
  pattern = ".Rmd",
  all.files = TRUE,
  include.dirs = TRUE,
  recursive = TRUE
)

 [1] "doc/additional.Rmd"                       "doc/ggbetweenstats.Rmd"                  
 [3] "doc/ggcoefstats.Rmd"                      "doc/ggcorrmat.Rmd"                       
 [5] "doc/gghistostats.Rmd"                     "doc/ggpiestats.Rmd"                      
 [7] "doc/ggscatterstats.Rmd"                   "doc/tests_and_coverage.Rmd"              
 [9] "README.Rmd"                               "vignettes/additional.Rmd"                
[11] "vignettes/ggbetweenstats.Rmd"             "vignettes/ggcoefstats.Rmd"               
[13] "vignettes/ggcorrmat.Rmd"                  "vignettes/gghistostats.Rmd"              
[15] "vignettes/ggpiestats.Rmd"                 "vignettes/ggscatterstats.Rmd"            
[17] "vignettes/tests_and_coverage.Rmd"         "vignettes/web_only/combine_plots.Rmd"    
[19] "vignettes/web_only/gallery.Rmd"           "vignettes/web_only/ggstatsplot_paper.Rmd"
[21] "vignettes/web_only/purrr_examples.Rmd"    "vignettes/web_only/session_info.Rmd"     
[23] "vignettes/web_only/theme_ggstatsplot.Rmd"

# sweave files
list.files(
  pattern = ".Rnw",
  all.files = TRUE,
  include.dirs = TRUE,
  recursive = TRUE
)

character(0)
```

**Comment by russHyde** [comment]
I just had a look at the source code for 1.0.3. There's a [statement ](https://github.com/jimhester/lintr/commit/7fe7e9093c7e9f41eebe600d89a98375d68f8f40#diff-6deea500c8accc924eac5b361a674941)missing in extract.R relative to the github source code. The current code checks whether there is at least one code-block before it checks the code. If there isn't there's a quick exit from extract_r_source.

**Comment by russHyde** [comment]
Continuing: the error is thrown because you are linting a file that either has no code blocks, or has code blocks that don't contain R code. I fixed a couple of bugs related to this, but the fixes have only been integrated into the master branch, they haven't yet been released to CRAN. I'm 'closing the issue for now; but if you try running lintr with the github version and the error persists, can you reopen this issue, please.
